core/unix/signal.c

```diff
@@ -7232,18 +7231,22 @@ notify_and_jmp_without_stack(KSYNCH_TYPE *notify_var, byte *continuation, byte *
         asm("mov " ASM_R1 ", #1");
         asm("str " ASM_R1 ",[" ASM_R0 "]");
         asm("ldr " ASM_R1 ", %0" : : "m"(continuation));
+        asm("ldr " ASM_R2 ", %0" : : "m"(xsp));
+        asm("mov " ASM_XSP ", " ASM_R2);
         asm("b dynamorio_condvar_wake_and_jmp");
 #endif
     } else {
         ksynch_set_value(notify_var, 1);
 #ifdef DR_HOST_NOT_TARGET
         ASSERT_NOT_REACHED();
 #elif defined(X86)
-        asm("mov %0, %%" ASM_XSP : : "m"(xsp));
         asm("mov %0, %%" ASM_XAX : : "m"(continuation));
+        asm("mov %0, %%" ASM_XSP : : "m"(xsp));
```

Comment:
Why was this reordering required for x86?

Reply:
That's me worried that gcc will use xsp to fetch the parameter (e.g., 32-bit, or maybe it put a copy on the stack and it's not still in a register and there's no frame pointer, etc.), and so it's safest to mess up xsp last.  I'll add a comment.

Reply:
Done.